libsolidity/codegen/ContractCompiler.cpp

```diff
@@ -604,6 +604,8 @@ bool ContractCompiler::visit(VariableDeclaration const& _variableDeclaration)
 
 bool ContractCompiler::visit(FunctionDefinition const& _function)
 {
+	solAssert(_function.isImplemented(), "");
```

Comment:
How is this guaranteed?

Reply:
I added it not because I know it's guaranteed but because I think it's expected by this function. It cannot generate code for an unimplemented function after all. So if this precondition is not met, we have a bug somewhere.

No test triggers that after my fix though so I think we're safe. If we are not, we'll at least get an ICE here, not somewhere deeper.

Reply:
Ok